js/huobi.js

```diff
@@ -2845,6 +2872,33 @@ module.exports = class huobi extends Exchange {
         return result;
     }
 
+    networkIdToCode (networkId, currencyCode = undefined) {
+        // here network-id is provided as a pair of currency & chain (i.e. trc20usdt)
+        if (currencyCode === undefined) {
```

Comment:
@ttodua can't we change `networkTitlesByJunctions` to be a direct map between networkId -> Unified Code?

It would be something like this:
```Javascript

{
    "bep20usdt", "BEP20",
    (...)
}
```
⬆️ I don't see any benefit in indexing by the currency, but I might be missing something. Additionally, by doing this, we can remove the `currencyCode` requirement from here, because if we only want to convert networkCode Id to unified (that does not hold any reference to the currency) we should not need the currency either I think

Reply:
I have thought of that in the beginning too, but I would cause discrepancy in the logic of `networkCodeToId` and `networkIdToCode`. 
for example, to use `networkCodeToId` you need to input `currencyCode` too which is mandatory for huobi: 
`networkCodeToId ('BEP20', 'USDT') ----> bep20usdt`
so, to make it working in reverse, i thought we should samely do:
`networkIdToCode  ('bep20usdt', 'USDT') ----> BEP20 `

but you are right, we can remove `currency` requirement from id->chain bindings.

Reply:
Sorry, I think I have incorrectly explained in above comment, let me explain in different way, how I build unified network junctions for huobi.

Reply:
I was thinking that I had explained this somewhere a few days ago, but I don't find my such explanation here. So, I will write-down. 
First, we should define the `networks` and `networksById` options, which should link exchange-specific networks to unified networks. That junction, in most cases of our existing implementations, are like `exchange_specific_id -> unified_network_code` (where `exchange_specific_id` is the id that is used in raw request to api endpoints (for withdraw, depositAddress, etc).  however...

CASE 1
=

 if we made the same approach for huobi, it will not be viable, because huobi's "network_specific_id" does not have not generic phrase, like `BEP20, ETH, ERC20, TRX`... or whatever, instead their in-use "network_specific_ids" are "variable", constructed using currency-code and network-title, i.e. `bep20usdt`, `bep20bnb`, `erc20bnb` and etc (_which is very unintelligent design ofc)_ . So, if we don't fix that garbage junctions, we would not be able to define these below properties 
```
'networksById':{ 
     'bep20usdt': 'BEP20',
     'bep20bnb': 'BEP20',
     'bep20usdt': 'BEP20',
     .... 
}
```
`networksById` can be somehow defined (without conflict) but it will need to have populated with thousands of such currency-specific junction names under our `options`. we shouldnt do that, but even if you say that we should do that, then ok, but beside it we have more problem - it's impossible to define `networks` property at all:
````
'networks':{
     'BEP20': '...what should be here.. ?  bep20usdt , bep20bnb, .. or which one from hundreds of bep20 based junctions
     ....
},
````

CASE 2
=
so, to stay in normal regime, I constructed the following approach : 
1) as mentioned in top post of this PR, each currency objects (in raw response) has additional "displayName" (not network-specific-id-what-is-used-in-request-for-withraw-deposit endpoint), like:
```
USDT: {
    'chain_id': 'bep20usdt',
    'displayName': 'Bep20',
},
BNB: {
    'chain_id': 'bep20bnb',
    'displayName': 'Bep20',
},
```
so, as you see in that example, `displayName` is same format (i.e. `Bep20`)  for "that-network-related" currencies too (as opposed to "id" junction which is variable for each currency). So, I've used that `displayName` to call them 'exchange-specific-network-titles' and defined them in options:
````
'networks': {
    'BEP20', 'Bep20',
    'ERC20', 'Erc20',
    ... //so, quite simple job to list our 10 unified network codes-to-id linkings
},
'networksById': {
    'Bep20': 'BEP20',
    'Erc20': 'ERC20',
    ... // so, as you see, as opposed to the `CASE-1`, we dont have to defined hundreds of network-id-to-code junctions here,  like `bep20usdt : BEP20` etc.
}
```
and that looks more aligned to our whole library. 

However, as already clear for you, huobi endpoints require not such network "title" (i.e. Bep20), but the awkward junction-ids, like `bep20usdt`. so, we had to override n`IdToCode` and n`CodeToId` methods, because:

A) we dont want `networkCodeToId ('BEP20', 'USDT')` returned network "title" (i.e. `Bep20`), but instead it should return `bep20usdt`. that will be the correct behavior, even though in options we had said `networks -> BEP20` has `Bep20` id (I know that might be non-ideal, but that's only way still, let me continue...)
B) we also dont want `networkIdToCode ('Bep20'...)` to use anywhere, because for response parsing, we everywhere get i.e. `bep20usdt` like values from raw responses, so we will be handling `networkIdToCode ('bep20usdt'...)` cases actually. so, again, we don't have `networksById->bep20usdt..` properties defined, and thus, we instead need to have some container which will tell us what network-code is for `bep20usdt`.

So, to address A & B cases, we needed those two exchange-wide properties (`networkJunctionsByTitles`,) and it will sort our all these burden from users sight (without defining those awkward currency+chainid lists in implementations) and I'm using overrided n`CodeToId / IdToCode` methods in huobi, which handles such scenarios, while `super.networkCodeToId/IdToCode` is also used in their body, but `super.n-code-id` methods just do a basic lookup from `options->networks/networksById` properties, which are meant to be 'default way' for most exchanges.

hope this explanation clarifies the case. 

(last note - you are right, I might simplify the specific technical thing you mentioned in above comment, however, the whole above logic will still hold true)

Reply:
@ttodua I understand your point, but as I said, I think we are overcomplicating it in the networkdID->networkCode translation; it is better to spend a few more bytes and keep things more transparent IMO

One more thing, can't we keep everything inside the "standard" options?

`networksById` -> would store the flat map as we discussed before (current `networkTitlesByJunctions`).

`networks` -> would store a map indexed by the currency (current `networkJunctionsByTitles` )

⬆️ since we override the methods using them, it should be ok to handle the different format of `networks` right?  

@ttodua wdyt?

Reply:
> @ttodua I understand your point, but as I said, I think we are overcomplicating it in the networkdID->networkCode translation; it is better to spend a few more bytes and keep things more transparent IMO

ok 👍 in networkId->networkCode translation I will strip out the need of currencyCode, as `networkId`'s are unique across while data.

> One more thing, can't we keep everything inside the "standard" options?
> 
> `networksById` -> would store the flat map as we discussed before (current `networkTitlesByJunctions`).
> 
> `networks` -> would store a map indexed by the currency (current `networkJunctionsByTitles` )
> 
> ⬆️ since we override the methods using them, it should be ok to handle the different format of `networks` right?

for better understanding of me, can you please give me an example of how you mean it to be?

Reply:
@ttodua 
>  for better understanding of me, can you please give me an example of how you mean it to be?

For instance usually, the `networks` map is something like this right?

```
networks: {
    "TRC20": "TRX"
}
```

But at huobi would be something like this ⬇️  (basically the current `networkJunctionsByTitles`)

```
networks: {
    "USDT": {
        "BEP20": "bep20usdt"
    }
}
```